### ISSUE TITLE
feat(select): Add select header to the md-select

### ISSUE THREAD

**Issue by Eernie** [angular/components#2812]
#### Bug, feature request, or proposal:
Feature request

Yesterday I filled a pull request ( #2797 ) for adding a search bar in the md-select component. As @jelbourn pointed out this wasn't the right way to add this feature, so now I'm willing to do it the right way. 
What's the best way to address this? Just implement the same functionality as material does it? 

That will mean something like this is possible:
``` html
<md-select placeholder="Favorite food">
    <md-select-header>
        <input [ngModel]="searchTerm" type="search" placeholder="Search for a vegetable..">
    </md-select-header>   
        <md-option *ngFor="let food of foods | filter " [value]="food.value">
            {{ food.viewValue }}
        </md-option>
</md-select>
```

**Comment by jelbourn** [comment]
The idea is pretty much the same as [the AngularJS version](https://material.angularjs.org/latest/demo/select). One of the things holding us back from working on this has been that we're not sure what the appropriate accessibility story for this feature is. The most common use case the the select header is a filter input, but the standard `role="listbox"` doesn't have anything like a associated filter. Adding it would require some research as to whether there is something we can do to make this interaction work for screen-reader users. 

As far as API and implementation details go, @kara will likely have some input.

**Comment by Eernie** [comment]
Is it possible that I port the AngularJS version for now and add the whole `role="listbox"` part later when this is clear?

**Comment by fxck** [comment]
@Eernie go for it

**Comment by esinek** [comment]
@Eernie, FYI, another use case of md-select-header is a check all / uncheck all (button(s) or tri-state checkbox) if the options are multi-select.

**Comment by exequiel09** [comment]
What's the status of this? Is this merged or not?

**Comment by JasonAHeron** [comment]
I was wondering if there was any progress on this?

**Comment by crisbeto** [comment]
@JasonAHeron the related PR (#3211) was blocked for a while until we decide how to handle the accessibility of the header. It should be unblocked once https://github.com/angular/material2/pull/6856 gets in.

**Comment by doronsever** [comment]
Any news on this issue?
It looks like #6856 was merged

**Comment by crisbeto** [comment]
There's a pending pull request at https://github.com/angular/material2/pull/7835.

**Comment by andplusblaw** [comment]
Hi I was wondering if 'mat-select-header' is available yet?

**Comment by jrood** [comment]
@andplusblaw unfortunately not yet. Please leave feedback at https://github.com/angular/material2/pull/7835 if you have any.

**Comment by jariwbh** [comment]
when can we expect this feature to be work ?

**Comment by andplusblaw** [comment]
@jrood thank you

**Comment by DenisSemionov** [comment]
Still no? More than 1 year passed

**Comment by irowbin** [comment]
Any progress on this one?

**Comment by sainisagar310** [comment]
+1

**Comment by vehansayvazi** [comment]
+1

**Comment by jelbourn** [comment]
Closing as a duplicate of #5697 because that one has more votes (even though this one is older)
